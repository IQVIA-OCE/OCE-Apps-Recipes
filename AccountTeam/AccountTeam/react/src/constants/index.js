import { Platform } from 'react-native';

export { SORT_OPTION_LIST, ACCESS_LEVEL_PICKLIST } from './sortOptionsList';
export { NAMESPACE, LOCALIZATION_NAMESPACE } from './namespacePrefix';
export { LOADING_STATUS } from './loadingStatus';
export const isIphone = !Platform.isPad && Platform.OS !== 'web';
