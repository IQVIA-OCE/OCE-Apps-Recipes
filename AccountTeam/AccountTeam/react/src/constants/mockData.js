import { NAMESPACE } from ".";
const customSettingsForTrue = [{ [`${NAMESPACE}DoNotUseTerritories255Field__c`]: true }];
const customSettingsForFalse = [{ [`${NAMESPACE}DoNotUseTerritories255Field__c`]: false }];
const territoryIds = [{ Territory2Id: '0MI4x0000000TLGGA2' }, { Territory2Id: '0MI4x0000000TLHGA2' }, { Territory2Id: '0MI4x0000000TLIGA2' }]
const territoryNameIds = [{ Id: '0MI4x0000000TLGGA2' }, { Id: '0MI4x0000000TLHGA2' }, { Id: '0MI4x0000000TLIGA2' }]
const territoryNames = [{ [`${NAMESPACE}Territories__c`]: ";DM - San Francisco 20D02;TM - SPC - Aurora 20A02T06;TM - SPC - Joliet 20A02T11;" }]
const teamMembers = [{
    Territory2: {
        AccountAccessLevel: "Read",
        Description: null,
        Name: "DM - San Francisco 20D02",
    },
    User: {
        City: null,
        Country: "US",
        Email: "test1@test.com",
        FirstName: "Partner",
        LastName: "Admin",
        Manager: null,
        Name: "Partner Admin",
        [`${NAMESPACE}LastSyncTime__c`]: null,
        Phone: null
    }
},
{
    Territory2: {
        AccountAccessLevel: "Edit",
        Description: null,
        Name: "TM - SPC - Aurora 20A02T06",
    },
    User: {
        City: null,
        Country: null,
        Email: "test2@test.com",
        FirstName: "Test",
        LastName: "User",
        Manager: null,
        Name: "Test User",
        [`${NAMESPACE}LastSyncTime__c`]: '2021-11-16T12:59:07.000+0000',
        Phone: "480"
    }
},
{
    Territory2: {
        AccountAccessLevel: "Edit",
        Description: null,
        Name: "TM - SPC - Aurora 20A02T06"
    },
    User: {
        City: "Chandler",
        Country: "United States",
        Email: "test3@test.com",
        FirstName: "test user 2",
        LastName: "user 2",
        Manager: null,
        Name: "test user 2",
        [`${NAMESPACE}LastSyncTime__c`]: null,
        Phone: "(480) 479-9000"
    }
}]
const searchTeamMemeberData = [
    {
        Territory2: {
            AccountAccessLevel: "Edit",
            Description: null,
            Name: "TM - SPC - Aurora 20A02T06",
        },
        User: {
            City: null,
            Country: null,
            Email: "test4@test.com",
            FirstName: "test user 1",
            LastName: "user 1",
            Manager: null,
            Name: "test user 1",
            [`${NAMESPACE}LastSyncTime__c`]: '2021-11-16T12:59:07.000+0000',
            Phone: "480"
        }
    },
    {
        Territory2: {
            AccountAccessLevel: "Edit",
            Description: null,
            Name: "TM - SPC - Aurora 20A02T06"
        },
        User: {
            City: "Chandler",
            Country: "United States",
            Email: "test4@test.com",
            FirstName: "test user 2",
            LastName: "user 2",
            Manager: null,
            Name: "test user 2",
            [`${NAMESPACE}LastSyncTime__c`]: null,
            Phone: "(480) 479-9000"
        }
    }];
const layoutData = {
    compactLayoutable: true,
    createable: true,
    custom: false,
    customSetting: false,
    deepCloneable: false,
    deletable: true,
    deprecatedAndHidden: false,
    extendedBy: null,
    extendsInterfaces: null,
    fields: [
        {
            label: "Account Access Level",
            length: 40,
            mask: null,
            maskType: null,
            name: "AccountAccessLevel",
            nameField: false,
            namePointing: false,
            nillable: false,
            permissionable: false,
            picklistValues: [
                {
                    active: true,
                    defaultValue: false,
                    label: "Read Only",
                    validFor: null,
                    value: "Read",
                },
                {
                    active: true,
                    defaultValue: false,
                    label: "Read/Write",
                    validFor: null,
                    value: "Edit",
                },
                {
                    active: true,
                    defaultValue: false,
                    label: "Owner",
                    validFor: null,
                    value: "All",
                },
            ],
        }
    ],
    implementsInterfaces: null,
    feedEnabled: false,
    queryable: true,
    defaultImplementation: null,
    labelPlural: "Meeting Members",
    networkScopeFieldName: null
};
export { territoryIds, customSettingsForTrue, customSettingsForFalse, territoryNames, teamMembers, searchTeamMemeberData, layoutData, territoryNameIds };