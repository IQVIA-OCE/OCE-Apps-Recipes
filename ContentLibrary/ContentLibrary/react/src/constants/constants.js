export const OWNED_BY_ME = 'ownedByMe';
export const SHARED_WITH_ME = 'sharedWithMe';
export const NO_CONNECTION = 'No Connection';