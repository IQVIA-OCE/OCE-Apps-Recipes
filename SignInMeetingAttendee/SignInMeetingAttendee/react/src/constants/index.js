import { Platform } from 'react-native';

export { NAMESPACE, userId, profileId, organizationId } from './namespacePrefix';
export const isIphone = !Platform.isPad;