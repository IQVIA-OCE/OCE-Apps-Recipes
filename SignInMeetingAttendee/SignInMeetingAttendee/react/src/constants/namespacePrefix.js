import { environment } from 'oce-apps-bridges';

export const NAMESPACE = environment.namespace();

export const userId = environment.userID();

export const profileId = environment.profileId();

export const organizationId = environment.organizationId();
