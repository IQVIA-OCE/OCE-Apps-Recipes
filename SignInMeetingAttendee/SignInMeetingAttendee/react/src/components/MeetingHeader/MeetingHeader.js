import React, { useEffect, useState } from 'react';
import { View, Text, StyleSheet, NativeModules, Alert } from 'react-native';
import { environment, localized } from "oce-apps-bridges"
import { Paragraph } from 'apollo-react-native'
import { DateTime } from 'luxon';
import { useSelector } from 'react-redux';
import { NAMESPACE } from '../../constants'


const MeetingHeader = () => {
    const [highlightPanel, sethighLightPanel] = useState([]);
    const { meetingDetails, meetingHighLightPanel } = useSelector((state) => state.meeting);
    useEffect(() => {
        if (meetingHighLightPanel[0]) {
            sethighLightPanel(meetingHighLightPanel[0].fieldItems);
        } else {
            NativeModules.ReachabilityBridge.networkReachabilityStatus().then(
                (networkStatusString) => {
                    if (networkStatusString === 'No Connection') {
                        const fieldItems = [{
                            label: localized('Name', 'Meeting Name'),
                            layoutComponents: [{
                                type: "Field",
                                value: "Name",
                                details: {
                                    name: "Name",
                                    type: "string"
                                }
                            }]
                        },
                        {
                            label: `${localized('Start', 'Start')}${localized('Date_time', 'Date/Time')}`,
                            layoutComponents: [{
                                type: `Field`,
                                value: `${NAMESPACE}StartDateTime__c`,
                                details: {
                                    name: `${NAMESPACE}StartDateTime__c`,
                                    type: "datetime"
                                }
                            }],
                        },
                        {
                            label: `${localized('InputDateRange_EndDate_DefaultLabel', 'End')}${localized('Date_time', 'Date/Time')}`,
                            layoutComponents: [{
                                type: "Field",
                                value: `${NAMESPACE}EndDateTime__c`,
                                details: {
                                    name: `${NAMESPACE}EndDateTime__c`,
                                    type: "datetime"
                                }
                            }],
                        },
                        {
                            label: `${localized('AttachmentCreatedByIdLabel', 'Created By')}`,
                            layoutComponents: [
                                {
                                    type: "Field",
                                    value: "CreatedById",
                                    details: {
                                        name: "CreatedBy",
                                        type: "reference",
                                        relationshipName: 'CreatedBy'
                                    }
                                },
                                {
                                    type: "Separator",
                                    value: ", ",
                                },
                                {
                                    type: "Field",
                                    value: "CreatedDate",
                                    details: {
                                        name: "CreatedDate",
                                        type: "datetime"
                                    }
                                },
                            ],
                        },
                        {
                            label: `${localized('Location', 'Location')}`,
                            layoutComponents: [{
                                type: "Field",
                                value: `${NAMESPACE}Location__c`,
                                details: {
                                    name: `${NAMESPACE}Location__c`,
                                    type: "string"
                                }
                            }],
                        }];
                        sethighLightPanel(fieldItems);
                    }
                }

            ).catch(
                (reason) => {
                    Alert.alert(
                        "Something Went Wrong",
                        `Handle ReachabilityBridge Rejected promise - ${reason}`,
                        [
                            { text: "OK" }
                        ]
                    );
                }
            );
        }
    }, [meetingHighLightPanel]);
    const renderHighLightPanelValues = (comp) => {
        if (comp.length === 1) {
            if (comp[0].details.type === 'datetime') {
                return (meetingDetails[0] && meetingDetails[0][comp[0].value]) ? DateTime.fromISO(meetingDetails[0][comp[0].value], {
                    zone: environment.timeZone()
                }).toFormat('d/M/yy, hh:mm:a') : ''
            } else {
                return (meetingDetails[0] && meetingDetails[0][comp[0].value]) ? meetingDetails[0][comp[0].value] : ''
            }
        }
        if (comp.length > 1) {
            let value = '';
            comp.map((fields) => {
                if (fields.details && fields.details.type === 'reference') {
                    value = meetingDetails[0] && meetingDetails[0][`${fields.details.relationshipName}.Name`] ? `${meetingDetails[0][`${fields.details.relationshipName}.Name`]}` : ''
                }
                if (fields.type === 'Separator') {
                    value = `${value}${fields.value}`
                }
                if (fields.details && fields.details.type === 'datetime') {
                    value = `${value}${(meetingDetails[0] && meetingDetails[0][fields.value]) ? DateTime.fromISO(meetingDetails[0][fields.value], {
                        zone: environment.timeZone()
                    }).toFormat('d/M/yy, hh:mm:a') : ''} `;
                }
            })
            return value
        }
    }
    const renderHightLightPanel = () => {
        return highlightPanel && highlightPanel.filter((filelds) => filelds.layoutComponents[0].value !== 'Name').map((field, index) => {
            const { layoutComponents } = field;
            return (
                <View style={styles.layoutContainer}>
                    <View style={index === highlightPanel.length - 2 ? styles.row : styles.meetingInfo}>
                        <Paragraph style={[styles.textCenter, styles.infoText]}>

                            {
                                renderHighLightPanelValues(layoutComponents)
                            }

                        </Paragraph>
                    </View>
                    <View style={styles.row}>
                        <Paragraph style={[styles.textCenter, styles.labelText]}>{field.label}</Paragraph>
                    </View>
                </View>
            )
        })
    }
    return (
        <View style={styles.headerContainer}>
            {renderHightLightPanel()}
        </View>
    )
};

export default MeetingHeader;

const styles = StyleSheet.create({
    headerContainer: {
        flexDirection: 'row',
        paddingHorizontal: 50,
        paddingVertical: 20,
    },

    layoutContainer: {
        flex: 1,
    },
    row: {
        flexDirection: 'row',
    },
    textCenter: {
        flex: 1,
        textAlign: 'center'
    },
    meetingInfo: {
        flexDirection: 'row',
        borderRightWidth: 2,
    },
    infoText: {
        fontWeight: 'bold',
        fontSize: 16,
        paddingHorizontal: 2
    },
    labelText: {
        fontSize: 14,
        color: '#8D8D8D',
        marginTop: 5
    }
});
