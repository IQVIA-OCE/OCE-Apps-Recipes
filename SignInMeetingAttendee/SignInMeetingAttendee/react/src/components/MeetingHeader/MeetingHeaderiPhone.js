import React, { useState, useEffect } from 'react';
import { View, TouchableOpacity, StyleSheet, NativeModules, Alert } from 'react-native';
import { Paragraph, secondaryBlue, IconButton } from 'apollo-react-native'
import { DateTime } from 'luxon';
import { useSelector } from 'react-redux';
import UserDetails from './UserDetails/UserDetails'
import { NAMESPACE } from '../../constants'
import { environment } from "../../../bridge/EnvironmentData/EnvironmentData.native"
import { localized } from '../../../bridge/Localization/localization.native';


const MeetingHeaderiPhone = ({ onSignInMeetingiPhone }) => {
    const [highlightPanel, sethighLightPanel] = useState([]);
    const { meetingDetails, meetingHighLightPanel } = useSelector((state) => state.meeting);
    useEffect(() => {
        if (meetingHighLightPanel[0]) {
            sethighLightPanel(meetingHighLightPanel[0].fieldItems);
        } else {
            NativeModules.ReachabilityBridge.networkReachabilityStatus().then(

                (networkStatusString) => {
                    if (networkStatusString === 'No Connection') {
                        const fieldItems = [{
                            label: localized('Name', 'Meeting Name'),
                            layoutComponents: [{
                                type: "Field",
                                value: "Name",
                                details: {
                                    name: "Name",
                                    type: "string"
                                }
                            }]
                        },
                        {
                            label: `${localized('Start', 'Start')}${localized('Date_time', 'Date/Time')}`,
                            layoutComponents: [{
                                type: `Field`,
                                value: `${NAMESPACE}StartDateTime__c`,
                                details: {
                                    name: `${NAMESPACE}StartDateTime__c`,
                                    type: "datetime"
                                }
                            }],
                        },
                        {
                            label: `${localized('InputDateRange_EndDate_DefaultLabel', 'End')}${localized('Date_time', 'Date/Time')}`,
                            layoutComponents: [{
                                type: "Field",
                                value: `${NAMESPACE}EndDateTime__c`,
                                details: {
                                    name: `${NAMESPACE}EndDateTime__c`,
                                    type: "datetime"
                                }
                            }],
                        },
                        {
                            label: `${localized('AttachmentCreatedByIdLabel', 'Created By')}`,
                            layoutComponents: [
                                {
                                    type: "Field",
                                    value: "CreatedById",
                                    details: {
                                        name: "CreatedBy",
                                        type: "reference",
                                        relationshipName: 'CreatedBy'
                                    }
                                },
                                {
                                    type: "Separator",
                                    value: ", ",
                                },
                                {
                                    type: "Field",
                                    value: "CreatedDate",
                                    details: {
                                        name: "CreatedDate",
                                        type: "datetime"
                                    }
                                },
                            ],
                        },
                        {
                            label: `${localized('Location', 'Location')}`,
                            layoutComponents: [{
                                type: "Field",
                                value: `${NAMESPACE}Location__c`,
                                details: {
                                    name: `${NAMESPACE}Location__c`,
                                    type: "string"
                                }
                            }],
                        }];
                        sethighLightPanel(fieldItems);
                    }
                }

            ).catch(
                // Log the rejection reason
                (reason) => {
                    Alert.alert(
                        "Something Went Wrong",
                        `Handle ReachabilityBridge Rejected promise - ${reason}`,
                        [
                            { text: "OK" }
                        ]
                    );
                }
            );
        }
    }, [meetingHighLightPanel]);

    const renderHighLightPanelValues = (comp) => {
        if (comp.length === 1) {
            if (comp[0].details.type === 'datetime') {
                return (meetingDetails[0] && meetingDetails[0][comp[0].value]) ? DateTime.fromISO(meetingDetails[0][comp[0].value], {
                    zone: environment.timeZone()
                }).toFormat('d/M/yy, hh:mm:a') : ''
            } else {
                return (meetingDetails[0] && meetingDetails[0][comp[0].value]) ? meetingDetails[0][comp[0].value] : ''
            }
        }
        if (comp.length > 1) {
            let value = '';
            comp.map((fields) => {
                if (fields.details && fields.details.type === 'reference') {
                    value = meetingDetails[0] && meetingDetails[0][`${fields.details.relationshipName}.Name`] ? `${meetingDetails[0][`${fields.details.relationshipName}.Name`]}` : ''
                }
                if (fields.type === 'Separator') {
                    value = `${value}${fields.value}`
                }
                if (fields.details && fields.details.type === 'datetime') {
                    value = `${value}${(meetingDetails[0] && meetingDetails[0][fields.value]) ? DateTime.fromISO(meetingDetails[0][fields.value], {
                        zone: environment.timeZone()
                    }).toFormat('d/M/yy, hh:mm:a') : ''} `;
                }
            })
            return value
        }
    }
    const renderHightLightPanel = () => {
        return highlightPanel && highlightPanel.map((field, index) => {
            const { layoutComponents } = field;
            return (
                <View style={styles.layoutContainer}>
                    <View style={styles.row}>
                        <Paragraph style={styles.infoText}>{field.label}</Paragraph>
                    </View>
                    <View style={styles.row}>
                        <Paragraph style={styles.labelText}>{renderHighLightPanelValues(layoutComponents)}</Paragraph>
                    </View>
                </View>
            )
        })
    }
    return (
        <View style={styles.headerContainer}>
            <UserDetails />
            <View style={styles.detailContiner}>
                <View>
                    <IconButton
                        icon="microphone"
                        disabled={true}
                        color={"#000"}
                        size={18}
                    />
                </View>
                <View style={{}}>
                    <View style={styles.layoutContainer}>
                        <Paragraph style={[styles.detailText]}>{localized("Global_Details", "Details")}</Paragraph>
                    </View>
                    <View>
                        {renderHightLightPanel()}
                    </View>
                </View >
            </View >
            <View style={styles.btnContainer}>
                <TouchableOpacity onPress={onSignInMeetingiPhone}>
                    <View style={styles.btn}><Paragraph style={styles.btnText}>{localized("Signature", "Sign")}</Paragraph></View>
                </TouchableOpacity>
            </View>
        </View >
    )
};

export default MeetingHeaderiPhone;

const styles = StyleSheet.create({
    headerContainer: {
        paddingHorizontal: 20,
        paddingVertical: 16,
    },

    detailContiner: {
        flexDirection: 'row',
        flex: 1,
        marginRight: 20
    },
    btnContainer: {
        alignContent: 'center',
        alignItems: 'center'
    },
    btn: {
        paddingHorizontal: 30,
        paddingVertical: 5,
        backgroundColor: secondaryBlue[400],
        borderRadius: 20
    },
    btnText: {
        color: 'white',
        fontWeight: 'bold'
    },
    layoutContainer: {
        marginTop: 10
    },
    row: {
        flexDirection: 'row',
    },
    detailText: {
        fontSize: 16,
        color: '#000',
        marginTop: 0,
        fontWeight: 'bold'
    },
    infoText: {
        fontWeight: 'bold',
        fontSize: 16,
    },
    labelText: {
        fontSize: 14,
        color: '#8D8D8D',
        marginTop: 0,
        marginRight: 10,
    }
});