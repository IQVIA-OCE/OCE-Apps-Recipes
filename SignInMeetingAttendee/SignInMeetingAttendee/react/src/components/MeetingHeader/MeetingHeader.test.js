import React from 'react';
import { NativeModules } from 'react-native';
import MeetingHeader from './MeetingHeader';
import { render } from '@testing-library/react-native';
import { useSelector, useDispatch } from 'react-redux';
import { records, meetingHighLightPanel } from "../../constants/mockData";

jest.useFakeTimers();


jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));


NativeModules.ReachabilityBridge = {
    networkReachabilityStatus: jest.fn()
        .mockResolvedValueOnce('ONLINE')
        .mockRejectedValue('ERROR')
}

describe('Meeting Header', () => {
    it('Meeting Header Component should render properly', () => {
        useDispatch.mockReturnValue(() => jest.fn());
        useSelector.mockImplementation((cb) => cb({
            meeting: {
                meetingDetails: records,
                meetingHighLightPanel: meetingHighLightPanel
            }
        }));
        const { getByText } = render(
            <MeetingHeader />
        );
        jest.advanceTimersByTime(1000);
        expect(getByText).toBeTruthy();
    })

});

