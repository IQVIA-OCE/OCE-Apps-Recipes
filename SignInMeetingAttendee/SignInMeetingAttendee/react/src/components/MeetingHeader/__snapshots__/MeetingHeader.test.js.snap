// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meeting Header Meeting Header Component should render properly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingHorizontal": 50,
      "paddingVertical": 20,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "borderRightWidth": 2,
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Array [
                  Object {
                    "flex": 1,
                    "textAlign": "center",
                  },
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "paddingHorizontal": 2,
                  },
                ],
              ],
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Array [
                  Object {
                    "flex": 1,
                    "textAlign": "center",
                  },
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "paddingHorizontal": 2,
                  },
                ],
              ],
            ],
          ]
        }
      >
        30/8/21, 11:35:AM
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Array [
                  Object {
                    "flex": 1,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#8D8D8D",
                    "fontSize": 14,
                    "marginTop": 5,
                  },
                ],
              ],
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Array [
                  Object {
                    "flex": 1,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#8D8D8D",
                    "fontSize": 14,
                    "marginTop": 5,
                  },
                ],
              ],
            ],
          ]
        }
      >
        StartDate/Time
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "borderRightWidth": 2,
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Array [
                  Object {
                    "flex": 1,
                    "textAlign": "center",
                  },
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "paddingHorizontal": 2,
                  },
                ],
              ],
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Array [
                  Object {
                    "flex": 1,
                    "textAlign": "center",
                  },
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "paddingHorizontal": 2,
                  },
                ],
              ],
            ],
          ]
        }
      >
        31/8/21, 11:35:AM
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Array [
                  Object {
                    "flex": 1,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#8D8D8D",
                    "fontSize": 14,
                    "marginTop": 5,
                  },
                ],
              ],
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Array [
                  Object {
                    "flex": 1,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#8D8D8D",
                    "fontSize": 14,
                    "marginTop": 5,
                  },
                ],
              ],
            ],
          ]
        }
      >
        EndDate/Time
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "borderRightWidth": 2,
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Array [
                  Object {
                    "flex": 1,
                    "textAlign": "center",
                  },
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "paddingHorizontal": 2,
                  },
                ],
              ],
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Array [
                  Object {
                    "flex": 1,
                    "textAlign": "center",
                  },
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "paddingHorizontal": 2,
                  },
                ],
              ],
            ],
          ]
        }
      >
        , 24/8/21, 11:35:AM 
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Array [
                  Object {
                    "flex": 1,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#8D8D8D",
                    "fontSize": 14,
                    "marginTop": 5,
                  },
                ],
              ],
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Array [
                  Object {
                    "flex": 1,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#8D8D8D",
                    "fontSize": 14,
                    "marginTop": 5,
                  },
                ],
              ],
            ],
          ]
        }
      >
        Created By
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Array [
                  Object {
                    "flex": 1,
                    "textAlign": "center",
                  },
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "paddingHorizontal": 2,
                  },
                ],
              ],
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Array [
                  Object {
                    "flex": 1,
                    "textAlign": "center",
                  },
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "paddingHorizontal": 2,
                  },
                ],
              ],
            ],
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Array [
                  Object {
                    "flex": 1,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#8D8D8D",
                    "fontSize": 14,
                    "marginTop": 5,
                  },
                ],
              ],
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Array [
                  Object {
                    "flex": 1,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#8D8D8D",
                    "fontSize": 14,
                    "marginTop": 5,
                  },
                ],
              ],
            ],
          ]
        }
      >
        Location
      </Text>
    </View>
  </View>
</View>
`;
